src/openrct2/scenario/scenario.c

```diff
@@ -566,19 +566,83 @@ static sint32 scenario_create_ducks()
  *
  * @return eax
  */
+#ifndef DEBUG_DESYNC 
 uint32 scenario_rand()
+#else
+static FILE *fp = NULL;
+static const char *realm = "LC";
+
+uint32 dbg_scenario_rand(const char *file, const char *function, const uint32 line, const void *data)
+#endif
 {
+    uint32 originalSrand0 = gScenarioSrand0;
+    gScenarioSrand0 += ror32(gScenarioSrand1 ^ 0x1234567F, 7);
+    gScenarioSrand1 = ror32(originalSrand0, 3);
+
 #ifdef DEBUG_DESYNC
+    if (fp == NULL)
+    {
+        if (network_get_mode() == NETWORK_MODE_SERVER)
+        {
+            fp = fopen("server_rand.txt", "wt");
+            realm = "SV";
+        }
+        else if (network_get_mode() == NETWORK_MODE_CLIENT)
+        {
+            fp = fopen("client_rand.txt", "wt");
+            realm = "CL";
+        }
+        else
+        {
+            if (fp)
+                fclose(fp);
+            fp = NULL;
+            realm = "LC";
+        }
+    }
+    if (fp)
+    {
+        fprintf(fp, "Tick: %d, Rand: %08X - REF: %s:%u %s (%p)\n", gCurrentTicks, gScenarioSrand1, file, line, function, data);
+    }
     if (!gInUpdateCode) {
         log_warning("scenario_rand called from outside game update");
-        assert(false);
+        //assert(false);
```

Comment:
@ZehM4tt any particular reason this is commented out? I mentioned on gitter I hit this condition at times, but the PRNG looks fine regardless. Are there known good paths leading to this point?

Can you either provide justification and remove this line or leave the assert in?